Converting angular workspace to nrwl while preserving layout fails to set default project

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->
When converting angular workspace to nrwl workspace while preserving layout using '--preserve-angular-cli-layout', the defaultProject property in nx.json is not set to existing angular project.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->
defaultProject property of nx.json should be set to existing angular project in the workspace.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->
1. ng new TourOfHeroes
2. ng add @nrwl/angular@14.5.10 --preserve-angular-cli-layout
3. Check nx.json
4. Unable to run 'npm start' without explicitly mentioning '--project' parameter

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->
Git URL: https://github.com/mazhar-fis/nx-issue-1.git

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->
Error: Both project and target have to be specified
    at parseRunOneOptions (C:\Projects\npx\temp\node_modules\nx\src\command-line\run-one.js:76:15)
    at Object.<anonymous> (C:\Projects\npx\temp\node_modules\nx\src\command-line\run-one.js:20:22)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Projects\npx\temp\node_modules\tslib\tslib.js:115:62)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! temp@0.0.0 start: `ng serve`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the temp@0.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\someone\AppData\Roaming\npm-cache\_logs\2022-09-30T05_11_33_802Z-debug.log

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
  NX   Report complete - copy this into the issue template

   Node : 14.19.3
   OS   : win32 x64
   npm  : 6.14.17

   nx : 14.5.10
   @nrwl/angular : 14.5.10
   @nrwl/cypress : 14.5.10
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.5.10
   @nrwl/eslint-plugin-nx : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.5.10
   @nrwl/js : 14.5.10
   @nrwl/linter : 14.5.10
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.5.10
   @nrwl/web : 14.5.10
   @nrwl/workspace : 14.5.10
   typescript : 4.7.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:

--- a/e2e/angular-core/src/ng-add.test.ts
+++ b/e2e/angular-core/src/ng-add.test.ts
@@ -148,6 +148,7 @@ describe('convert Angular CLI workspace to an Nx workspace', () => {
       cli: {
         packageManager: packageManager,
       },
+      defaultProject: project,
       implicitDependencies: {
         '.eslintrc.json': '*',
         'package.json': {

--- a/packages/angular/src/generators/ng-add/migrate-from-angular-cli.spec.ts
+++ b/packages/angular/src/generators/ng-add/migrate-from-angular-cli.spec.ts
@@ -243,9 +243,7 @@ describe('workspace', () => {
 
     it('should set the default project correctly', async () => {
       await migrateFromAngularCli(tree, {});
-      expect(readJson(tree, 'nx.json').defaultProject).toMatchInlineSnapshot(
-        `"myApp"`
-      );
+      expect(readJson(tree, 'nx.json').defaultProject).toBe('myApp');
     });
 
     it('should create nx.json', async () => {

--- a/packages/angular/src/generators/ng-add/migrate-from-angular-cli.spec.ts
+++ b/packages/angular/src/generators/ng-add/migrate-from-angular-cli.spec.ts
@@ -241,6 +241,13 @@ describe('workspace', () => {
       ).not.toBeDefined();
     });
 
+    it('should set the default project correctly', async () => {
+      await migrateFromAngularCli(tree, {});
+      expect(readJson(tree, 'nx.json').defaultProject).toMatchInlineSnapshot(
+        `"myApp"`
+      );
+    });
+
     it('should create nx.json', async () => {
       await migrateFromAngularCli(tree, { defaultBase: 'main' });
       expect(readJson(tree, 'nx.json')).toMatchSnapshot();

--- a/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
+++ b/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
@@ -3,8 +3,10 @@ import {
   formatFiles,
   installPackagesTask,
   readJson,
+  readWorkspaceConfiguration,
   Tree,
   updateJson,
+  updateWorkspaceConfiguration,
 } from '@nrwl/devkit';
 import { nxVersion } from '../../utils/versions';
 import type { GeneratorOptions } from './schema';
@@ -35,6 +37,10 @@ export async function migrateFromAngularCli(
   validateWorkspace(tree);
   const projects = getAllProjects(tree);
   const options = normalizeOptions(tree, rawOptions, projects);
+
+  const defaultProject = projects.apps.find((app) =>
+    app.config.sourceRoot?.startsWith('src')
+  );
 
   if (options.preserveAngularCliLayout) {
     addDependenciesToPackageJson(
@@ -101,6 +107,14 @@ export async function migrateFromAngularCli(
     }
 
     await formatFiles(tree);
+  }
+
+  if (defaultProject) {
+    const workspaceConfig = readWorkspaceConfiguration(tree);
+    updateWorkspaceConfiguration(tree, {
+      ...workspaceConfig,
+      defaultProject: defaultProject.name,
+    });
   }
 
   if (!options.skipInstall) {

--- a/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
+++ b/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
@@ -38,9 +38,7 @@ export async function migrateFromAngularCli(
   const projects = getAllProjects(tree);
   const options = normalizeOptions(tree, rawOptions, projects);
 
-  const defaultProject = projects.apps.find((app) =>
-    app.config.sourceRoot?.startsWith('src')
-  );
+  const defaultProject = projects.apps.find((app) => app.config.root === '');
 
   if (options.preserveAngularCliLayout) {
     addDependenciesToPackageJson(

